src/runtime/path_resolver.ts

```diff
@@ -0,0 +1,169 @@
+import { ApiRoute, PathPair } from "./mod.ts";
+
+export interface QualifiedPathPair {
+    module: string;
+    route: string;
+    path: PathPair;
+};
+
+
+/**
+ * A class to resolve paths to their associated {@link QualifiedPathPair}.
+ * 
+ * This is used to look up `QualifiedApiRoute` values by the requested path.
+ */
+export class PathResolver {
+    private readonly exactRoutes: Map<string, QualifiedPathPair> = new Map();
+    private readonly prefixRoutes: Map<string, QualifiedPathPair> = new Map();
+
+    private readonly _collisions: Set<QualifiedPathPair> = new Set();
+
+    public constructor(routes: QualifiedPathPair[]) {
+        this._collisions = PathResolver.getCollisions(routes);
```

Comment:
collisions should be checked at build time and routes should be speedy at runtime. assume everything is valid at runtime.

i see why you were concerned about perf

Reply:
The `PathResolver` is built once at dev server startup

Reply:
Now only checks at build time, and is assumed to be correct at runtime

Reply:
see my comment in the other thread about serverless